docs/src/backup_recovery.md

```diff
@@ -89,6 +95,26 @@ The destination path can be any URL pointing to a folder where
 the instance can upload the WAL files, e.g.
 `s3://BUCKET_NAME/path/to/folder`.
 
+#### IAM Role for Service Account (IRSA)
+
+In order to use IRSA we need an `annotation` in the cluster's serviceaccount.
+Using the `inheritedMetadata` configuration, the metadatas are propagated to **all** the resources created by the operator for a given cluster.
```

Comment:
Oh I didn't know that the annotations is kept. I definitely prefer this approach too. Let me check that, if that works as you say I would prefer not to talk about the `inheritedMetadata` option at all.
Does it make sense?

Reply:
Don't you think we could change the spec a bit to add a new parameter (because that'd be painful to ensure that the serviceaccount is created before the `Cluster` resource in a gitops workflow) ?:

```yaml

serviceAccount:
  annotations: {...}
```

Reply:
Thanks, I prefer this option too (I mean creating before the `ServiceAccount` with the required annotation) and I also would not mention the `spec.inheritedMetadata` option to achieve this or at least I would just say that is also possible, if preferred for any reasons.
Regarding the additional spec field, this is an option too but I also think that with some GitOps flow we can rely on some kind of dependencies, like for example Kustomize does with `spec.dependsOn` into the `Kustomization` resource. However, if you are having a strong option for that additional spec field, we can ask for some maintainers opinion here to get some additional feedback!